var/spack/repos/builtin/packages/ncurses/package.py

```diff
@@ -40,6 +40,7 @@ class Ncurses(AutotoolsPackage, GNUMirrorPackage):
 
     patch('patch_gcc_5.txt', when='@6.0%gcc@5.0:')
     patch('sed_pgi.patch',   when='@:6.0')
+    patch('ncurses-5.9-clang.patch', when='@:5.9%clang')
```

Comment:
Did you create this patch yourself? Or did you get it somewhere? If the latter, can you include a link?

Also, what clang version did you try this with? An older one or a newer one? Or do you expect that all clang versions have this issue?

Reply:
Here is an instance of the patch. I got it somewhere else, but it is the same.
https://github.com/CartoDB/omnibus-cartodb/blob/master/config/patches/ncurses/ncurses-clang.patch

I tested with clang 10.